be/src/vec/functions/array/function_array_reduce.cpp

```diff
@@ -0,0 +1,206 @@
+
+    DataTypePtr get_return_type_impl(const ColumnsWithTypeAndName & arguments) const override {
+        DataTypes argument_types(arguments.size() - 1);
+        const DataTypeArray * arg = check_and_get_data_type<DataTypeArray>(arguments[1].type.get());
+        argument_types[0] = arg->get_nested_type();
+
+        columnWithTypeAndName = arguments[1];
+        
+
+        if (!aggregate_function)
+        {
+            aggregate_function = vectorized::AggregateFunctionSimpleFactory::instance().get(
+                    "sum", argument_types);
+        }
+
+        std::cout << "YEAH" << std::endl;
+        
+        return arguments[1].type;
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        // 获取函数名                        
+        const ColumnConst& rhs_value_column =
+                static_cast<const ColumnConst&>(*block.get_by_position(arguments[0]).column.get());
+
+        // String aggregate_function_name_with_params = rhs_value_column.get_value<String>();
+        // std::cout << aggregate_function_name_with_params << std::endl;
+
+        ColumnPtr haystack_ptr = block.get_by_position(arguments[0]).column;
+
+        const ColumnConst* col_haystack_const =
+                check_and_get_column_const<ColumnString>(&*haystack_ptr);
+
+        // String aggregate_function_name_with_param = col_haystack_const->get_value<String>();
+        
+        // std::cout << aggregate_function_name_with_param << std::endl;
+
+        // AggregateFunctionSimpleFactory factory = AggregateFunctionSimpleFactory::instance();
+        // auto agg_function = factory.get(aggregate_function_name_with_params, {block.get_data_type(1)});
+
+        std::cout << "----" << std::endl;
+        // const IAggregateFunction & agg_func = *agg_function;
+        
+        const size_t num_arguments_columns = arguments.size() - 1;
+
+        std::vector<const IColumn *> aggregate_arguments_vec(num_arguments_columns);
+
+        std::cout << "----" << std::endl;
+        // for (size_t i = 0; i < num_arguments_columns; ++i) {
+            // ->convert_to_full_column_if_const();
+        // const IColumn * col = block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
+        const auto& col = block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
+        
+        if (const ColumnArray * arr = check_and_get_column<ColumnArray>(*col)) {
+            aggregate_arguments_vec[0] = &arr->get_data();
+            std::cout << "YEAH" << std::endl;
+            const auto& name = arr->get_name();
+            std::cout << &name << std::endl;
+            std::cout << "----" << std::endl;
+            // offsets_i = &arr->get_offsets();
+        }
+        else {
+            std::cout << "---------ERROR----------" << std::endl;
+        }
+
+        // if (i == 0)
```

Comment:
why not delete？

Reply:
I'm sorry, I submitted a PR to have someone look at my code, so there's a lot of problematic code in there.